x-pack/plugins/task_manager/server/task_claimers/strategy_mget.ts

```diff
@@ -0,0 +1,366 @@
+  ConcreteTaskInstance,
+  TaskStatus,
+  ConcreteTaskInstanceVersion,
+} from '../task';
+import { TASK_MANAGER_TRANSACTION_TYPE } from '../task_running';
+import {
+  isLimited,
+  TASK_MANAGER_MARK_AS_CLAIMED,
+  TaskClaimingBatches,
+} from '../queries/task_claiming';
+import { TaskClaim, asTaskClaimEvent, startTaskTimer } from '../task_events';
+import { shouldBeOneOf, mustBeAllOf, filterDownBy, matchesClauses } from '../queries/query_clauses';
+
+import {
+  IdleTaskWithExpiredRunAt,
+  InactiveTasks,
+  RunningOrClaimingTaskWithExpiredRetryAt,
+  SortByRunAtAndRetryAt,
+  EnabledTask,
+  OneOfTaskTypes,
+  RecognizedTask,
+} from '../queries/mark_available_tasks_as_claimed';
+
+import { TaskStore, SearchOpts } from '../task_store';
+import { isOk, asOk } from '../lib/result_type';
+
+interface OwnershipClaimingOpts {
+  claimOwnershipUntil: Date;
+  size: number;
+  taskTypes: Set<string>;
+  removedTypes: Set<string>;
+  excludedTypes: Set<string>;
+  taskStore: TaskStore;
+  events$: Subject<TaskClaim>;
+  definitions: TaskTypeDictionary;
+  taskMaxAttempts: Record<string, number>;
+}
+
+const SIZE_MULTIPLIER_FOR_TASK_FETCH = 4;
+
+export function claimAvailableTasksMget(opts: TaskClaimerOpts): Observable<ClaimOwnershipResult> {
+  const taskClaimOwnership$ = new Subject<ClaimOwnershipResult>();
+
+  claimAvailableTasksApm(opts)
+    .then((result) => {
+      taskClaimOwnership$.next(result);
+    })
+    .catch((err) => {
+      taskClaimOwnership$.error(err);
+    })
+    .finally(() => {
+      taskClaimOwnership$.complete();
+    });
+
+  return taskClaimOwnership$;
+}
+
+async function claimAvailableTasksApm(opts: TaskClaimerOpts): Promise<ClaimOwnershipResult> {
+  const apmTrans = apm.startTransaction(
+    TASK_MANAGER_MARK_AS_CLAIMED,
+    TASK_MANAGER_TRANSACTION_TYPE
+  );
+
+  try {
+    const result = await claimAvailableTasks(opts);
+    apmTrans.end('success');
+    return result;
+  } catch (err) {
+    apmTrans.end('failure');
+    throw err;
+  }
+}
+
+async function claimAvailableTasks(opts: TaskClaimerOpts): Promise<ClaimOwnershipResult> {
+  const { getCapacity, claimOwnershipUntil, batches, events$, taskStore } = opts;
+  const { definitions, unusedTypes, excludedTaskTypes, taskMaxAttempts } = opts;
+  const { logger } = opts;
+  const loggerTag = claimAvailableTasksMget.name;
+  const logMeta = { tags: [loggerTag] };
+  const initialCapacity = getCapacity();
+  const stopTaskTimer = startTaskTimer();
+
+  const removedTypes = new Set(unusedTypes); // REMOVED_TYPES
+  const excludedTypes = new Set(excludedTaskTypes); // excluded via config
+  const { docs, versionMap } = await searchAvailableTasks({
+    definitions,
+    taskTypes: new Set(definitions.getAllTypes()),
+    excludedTypes,
+    removedTypes,
+    taskStore,
+    events$,
+    claimOwnershipUntil,
+    size: initialCapacity * SIZE_MULTIPLIER_FOR_TASK_FETCH,
+    taskMaxAttempts,
+  });
+
+  if (docs.length === 0) return emptyClaimOwnershipResult();
```

Comment:
seems like we should return timing numbers here too even if no docs were claimed right?

Reply:
changing to return the timing numbers, thx!